### DISCUSSION TITLE
Internship and Job Opportunities

### DISCUSSION THREAD

**Discussion by kiranmai1101** [snigi-gupta/AttendingUniversityAtBuffalo#8] [General]
How are the internships and job opportunities at UB for Computer Science graduates considering its location?

**Comment by snigi-gupta** [comment]
Hi @kiranmai1101,

**TLDR; Jump to the 'note'**

Before I answer that question here are a few things you must know in order to understand internships and jobs in the US. There is a huge shift from the way internships and jobs work in India.

- The educational university is not responsible for getting you an internship/job in the US. In short, there are not TnP cells. There are career services aid but they are just there to help you in your search (like reviewing your resume, clearing your doubts, and telling you the search or interview process).
- It is easier to get a full-time job than to get an internship in the US for international students. Internships are very competitive and not everyone gets an internship. 70-80% of international students don't get a paid internship. 

Now to answer your question, as I mentioned UB will not play any role in getting you an internship or job. As the matter of fact, no university in the US plays any role. Some universities like Ivy League universities indirectly help students since they have a brand name. But if you are not graduating from an Ivy League then your university does not matter. Whether you graduate from the University of Southern California, Arizona State University, University at Buffalo, or San Jose State University (I took these as examples)- they are all the same in terms of getting you employment. How? because they have 0 contributions in your employment. Yes, they do differ in course structure, research opportunities, and tuition and fees.  So if you want to finalize a university and you have an option between UB vs some other university, consider ROI, course structure, research opportunities, class size etc. I can help you compare the two universities if that is the case.

If that is not the case, and you are wanting to know how to get an internship/job then here are some of the many resources:
- apply early usually 7-8 months before.
- impressive resume
- networking
- cold applications/ cold emails
- handshake
- self-projects
- interview preparation

To give you some idea:
For my internship, I submitted 400+  applications and I converted 2 internships. And that is an average number of applications.
For my job I submitted 500+ applications and I converted 4 offer letters.  A major reason for this was my networking and leadership skills.


Also, post COVID, location has become irrelevant, since interviews are now being conducted virtually and a company cannot discard a candidate just because he/she is on the other side of the country. I think this is something you shouldn't worry about now. Also, East cost has many opportunities too.

On the East side (NYC, Boston, Chicago) are some of the good/great cities to work in the tech domain. EAST is usually not preferred by Asians because it is a colder region. This is a personal choice and opinion.
On the West side there are plenty of opportunities and companies. It is called the silicon valley for a reason. That being said, many do not prefer WEST because expenses and taxes are also very high but that is a personal choice again.

**NOTE**: You will get an internship or a job only on the basis of your MERIT. By merit, I mean your core fundamentals and knowledge of the subject and domain. Your GPA will not matter. How you perform in your interviews ONLY matters and if you prove to them you are worth to their company they will hire you irrespective of your grade, university, location. That is how it works here in the US.

I hope I answered some of your questions. If you do have follow-up questions, feel free to post them here in this thread. I will reply in 2-3 days.

**Comment by vivekragi** [comment]
@snigi-gupta , I saw that you have applied for more than 400 companies to get an internship. Can you tell me from your experience, what care should be taken while applying for a company to ensure that we do not get ghosted by it?

In addition, can we apply for SDE-2/Senior Engineer roles and crack it if we have few years of experience under our belt?

**Reply by snigi-gupta** [reply]
@vivekragi ,
YOU WILL BE GHOSTED 100s OF TIMES! It's normal.
That shouldn't refrain you from taking your shot at the opportunity. Apply anyway, even if you feel there is a 1% chance. 
I applied 700 companies during my job hunt. Don't worry about all this. JUST APPLY.

You can apply for senior roles if you have 3+ years of experience . These interviews are very demanding, so make sure that you are prepared enough. Its not easy to crack these. I don't suggest applying to these roles unless you have 4+ years of experience, because the job is very demanding, they expect you to know a lot in advance when you join, and failure of not performing to those expectation can lead to getting laid off.

**Comment by poojitha66** [comment]
Hi @snigi-gupta 

I hope you're doing well! 

I wanted to seek your guidance regarding applying to internships for beginners. 

I am an incoming student (Fall-2023) at UB pursuing a Masters in Computer Science, I wanted to seek your guidance regarding internships for beginners. Could you please advise me on how to apply for internships and which job roles would be suitable for someone starting out? 

Additionally, I would appreciate any insights on when the ideal time to apply for internships would be while studying. 
Thanking you in advance for your help! :)

Best regards,
Poojitha

**Reply by snigi-gupta** [reply]
Hi @poojitha66 , 

Since you are about to start your education journey I would advise you the following:
### Things you can start doing before your classes start
1. Prepare your resume and make it ATS friendly - NOW 
2. Start leetcoding
3. Start applying to smaller companies or start-ups for internships. If you are getting too many rejects, tweak your resume again, and ensure it has the keywords for the parsers. 
4. Start networking NOW
5. Brush up basics and dive into the habit of learning 1 algorithm or data structure a day

### Things you should be doing during your classes (apart from studying for your courses)
1. Continue networking and look out for Grace Hopper Celebration and Society of Women Engineering conferences. Try to get tickets for these and attend their job fairs. 
2. Ideal time to start applying for internships is 1 year or 8-9 months in advance. So for Summer 2024 internship, you need to start applying latest by Fall 2023. 
3. Continue leetocoding and start preparing for interviews. 
4. Start adding/building your portfolio. Add projects, ideas, and assignments on your portfolio, and website.

### How to apply/find internships.
__Option 1__
1. Go on LinkedIn and look for internships in your domain/field. 
2. Cold apply to these (10-20 applications a day). Don't 'Easy Apply' on LinkedIn. Instead go to google, type `company_name careers` and then look for the positions there.

__Option 2__
1. Leverage the networking you did so far. Ask for referrals
2. The way to do this is, find the job link, share your PDF resume with the referrer. Follow back with them in 3 days, 5 days, and 7 days. If they don't respond, give up. If they do, follow up with them in 2 weeks, to personally share your resume with the recruiter or the hiring manager. 

__Option 3__
1. Reach out to the recruiter or the hiring manager via LinkedIn. And hope they respond back. 
2. You need to be active on LinkedIn so you appear more in recruiter/HM's search results. 

I hope this helps! Feel free to ask more questions if you have them! All the best with your search!
